_test/test_sqw/test_data_sqw_dnd.m

```diff
@@ -98,25 +98,133 @@
+            %
+            proj0 =  obj.ref_sqw.data.get_projection();
+            pix_range = obj.ref_sqw.data.pix.pix_range;
+            img_range = obj.ref_sqw.data.img_db_range;
+            full_pix_range = expand_box(pix_range(1,:),pix_range(2,:));
+            eval_img_range = proj0.transform_pix_to_img(full_pix_range);
+            full_img_range = expand_box(img_range(1,:),img_range(2,:));
+            
+            assertElementsAlmostEqual(eval_img_range,full_img_range);
+            
+            full_img_range = expand_box(img_range(1,:),img_range(2,:));
+            eval_pix_range = proj0.transform_img_to_pix(full_img_range);
+            assertElementsAlmostEqual(eval_pix_range,full_img_range);
+        end
+        %
         function test_loadobj_v0(obj)
             proj.u = [1,0,0];
-            proj.v = [0,1,0];            
+            proj.v = [0,1,0];
             ref_obj = data_sqw_dnd(proj,[1,0.01,2],[-1,1],[0,1],[0,1,10]);
             ld = load('data_sqw_dnd_V0_ref_data.mat');
             assertEqual(ref_obj,ld.obj);
             
             % check modern loader (if saved)
             obj.assertEqualWithSave(ref_obj);
         end
-        function test_get_proj_hkl_3D(obj)            
+        
+        
+        function test_get_proj_hkl_from_cut(obj)
+            proj = projection([1,1,0],[1,-1,0]);
+            proj.type='rrr';
+            % Be sure that cut is within the ranges of the real pixel range
+            % and cuts some pixels from all sides for the cut limits to be
+            % around real pix limits
+            source_cut = cut_sqw(obj.ref_sqw,proj,[-1,0.02,3],[-2,0.02,2],[-1,1],[-4,4]);
+            % Check projection 0;
+            proj_0 =  source_cut.data.get_projection();
+            assertTrue(isa(proj_0 ,'aProjection'));
+            img_range = source_cut.data.img_db_range;
+            ref_img_range = [ -1.0100   -2.0100   -1.0000   -4.0000;
+                3.0100    2.0100    1.0000    4.0000]; % actually look at cut ranges
+            assertElementsAlmostEqual(ref_img_range,img_range,'absolute',9.e-5);
+            
+            full_img_range = expand_box(img_range(1,:),img_range(2,:));
+            full_pix_img_range = proj_0.transform_img_to_pix(full_img_range);
+            eval_pix_range = [min(full_pix_img_range,[],2),max(full_pix_img_range,[],2)]';
+            ref_eval_range =  [-7.1945   -7.6191   -3.1416   -4.0000;...
+                14.0451   12.6648    3.1416    4.0000];
+            assertElementsAlmostEqual(ref_eval_range,eval_pix_range,'absolute',9.e-5);
+            real_pix_range = source_cut.data.pix.pix_range;
+            pix_ref_range = [-0.1238   -6.7724   -3.0949   -4.0000;...
+                5.6506   11.9775    3.0949    4.0000];
+            assertElementsAlmostEqual(pix_ref_range,real_pix_range,'absolute',9.e-5);
+            
+            %             %
+            %             % visualise correct image ranges if requested .

```

Comment:
Debugging left in?

Reply:
I think, its necessary to leave this code for future debugging. pix_ref_range is hardcoded and may change (I believe, it is still some unnecessary add_border lying around). And the only  way to understand that new range is correct, is to look at the pictures and see if they make a sense. This task becomes non-trivial task when you in couple of years time forgot what this test is all about.